libs/messaging/messaging-impl/src/main/kotlin/net/corda/messaging/mediator/processor/EventProcessor.kt

```diff
@@ -71,24 +72,32 @@ class EventProcessor<K : Any, S : Any, E : Any>(
         output.forEach { message ->
             val destination = messageRouter.getDestination(message)
             if (destination.type == RoutingDestination.Type.ASYNCHRONOUS) {
+                // Kafka - Add the request to the queue, so it can be processed in due course
                 consumerProcessorState.asynchronousOutputs.compute(key) { _, value ->
                     val list = value ?: mutableListOf()
                     list.add(message)
                     list
                 }
             } else {
+                // Http - Send the request immediately. Once the response arrives convert it to a kafka record and
+                // add it to the queue, so it can be processed in due course
                 @Suppress("UNCHECKED_CAST")
                 val reply = with(destination) {
                     message.addProperty(MessagingClient.MSG_PROP_ENDPOINT, endpoint)
                     client.send(message) as MediatorMessage<E>?
                 }
                 if (reply != null) {
+                    // Convert response to a record and add it to the queue
+                    @Suppress("UNCHECKED_CAST")
```

Comment:
With the suggested change to addTraceContextToRecord we can remove the cast:

```suggestion
                    // Convert response to a record and add it to the queue
                    
```